### PR TITLE
[NestedTensor] Integrate the mean operator along the jagged dimension into NestedTensor

### PR DESCRIPTION
**Pull request by jananisriram**
Summary:
Modify the existing `mean` operator in PyTorch, invoked by `torch.mean`, to allow for reductions along the jagged dimension of a nested tensor. The function originally had a basic implementation for reducing along 1 non-ragged dimension. This diff enables PyTorch users to invoke `torch.mean` on a nested tensor when reducing along the ragged dimension, e.g. `*` in a `(B, *, M)` nested tensor.

Parametrize unit tests from `sum` to verify the accuracy of the ragged reduction implementation for `torch.mean`. Add unit tests and parametrize `sum` unit tests to verify error handling for unsupported features in `NestedTensor` `torch.mean`.

Test Plan:
Verify that the new unit test passes via the following command:
```
buck2 run mode/{opt,inplace} //caffe2/test:nested -- --regex test_mean
```

```
buck2 run mode/{opt,inplace} //caffe2/test:nested -- --regex test_jagged_op
```

Differential Revision: D59654668

### CODE REVIEW

**Review by davidberard98 — APPROVED** [review]
Tests need to be fixed (or possibly skipped in this case), otherwise LGTM!

**Review by jbschlosser — APPROVED** [review]
Thanks for the PR! It'll be super nice to have this :)
